packages/assets-controllers/src/TokenBalancesController.ts

```diff
@@ -309,6 +344,72 @@ export class TokenBalancesController extends StaticIntervalPollingController<Tok
     );
   }
 
+  async #handleTokensControllerStateChange({
+    chainIds,
+  }: { chainIds?: Hex[] } = {}) {
+    const currentTokenBalancesState = this.messagingSystem.call(
+      'TokenBalancesController:getState',
+    );
+    const currentTokenBalances = currentTokenBalancesState.tokenBalances;
+    const currentAllTokens = this.#allTokens;
+    const chainIdsSet = new Set(chainIds);
+
+    // first we check if the state change was due to a token being removed
+    for (const currentAccount of Object.keys(currentTokenBalances)) {
```

Comment:
WOW, our objects are difficult to loop/select through! 3 layers of for loops lol.

This is fine as is, I do wonder if there is a way we can process and flatten this to make it a little easier to select through.